File: frontend/src/types/matterSearch.ts
export interface MatterSummary {
  matter_key: string;
  client_name: string;
  matter_name: string;
  practice_area: string | null;
  matter_type: string | null;
  matter_status: string;
  primary_attorney: string | null;
}

export interface MatterSearchResponse {
  total: number;
  limit: number;
  offset: number;
  matters: MatterSummary[];
}

export interface MatterAssignmentResponse {
  email_id: string;
  status: 'assigned' | 'already_assigned';
  matter_key: string | null;
  processing_status: string;
}

File: backend/app/schemas/matter_search.py
"""
Pydantic schemas for the Matter Search endpoint.

GET /api/matters — collection-level, read-only.
"""

from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel


class MatterSummary(BaseModel):
    """Lightweight Matter summary used for search results."""

    matter_key: str
    client_name: str
    matter_name: str
    practice_area: Optional[str]
    matter_type: Optional[str]
    matter_status: str
    primary_attorney: Optional[str]


class MatterSearchResponse(BaseModel):
    """Paginated response for GET /api/matters."""

    total: int
    limit: int
    offset: int
    matters: List[MatterSummary]
